AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -215,7 +219,8 @@ public final class APIClient: APIClientProtocol {
         return HTTPResponse(
             headers: result.headers,
             statusCode: result.statusCode,
-            responseBody: DownloadResponse(url: result.url)
+            responseBody: DownloadResponse(url: result.url),
+            responseData: Data()
```

Comment:
Can we have `result.data`?

Reply:
Good catch yeah thanks. `result` here is of type `URLSessionDownloadSuccess` so I've updated that object to fetch the data from the local downloaded file location in it's initialiser, so it will be available to return here then. I've updated the tests to validate this flow also

Reply:
Just a thought:

We are already providing URL, so not sure if simply passing `Data` make sense for "Download" operation - it could be a very big file..
Unfortunately, swift doesn't have a nice "data-on-demand" type (like [FileStream](https://www.morgandavison.com/2020/02/05/streaming-file-reader-in-swift/))

Reply:
Based on your suggestion below this now returns an object `HTTPResponse` which hasn't got any responseData field. so we wont return the file in this case. feels a bit safer i think